apps/api/src/__tests__/e2e_v1_withAuth/index.test.ts

```diff
@@ -1002,4 +1003,119 @@ describe("E2E Tests for v1 API Routes", () => {
       60000
     ); // 60 seconds
   });
+
+  describe("POST /v1/batch/scrape", () => {
+    it.concurrent("should require authorization", async () => {
+      const response: ScrapeResponseRequestTest = await request(TEST_URL)
+        .post("/v1/batch/scrape")
+        .send({ urls: ["https://firecrawl.dev"] });
+      expect(response.statusCode).toBe(401);
+    });
+
+    it.concurrent("should throw error for blocklisted URL", async () => {
+      const scrapeRequest: BatchScrapeRequest = {
+        urls: ["https://mendable.ai", "https://www.googel.com", "https://facebook.com/fake-test"],
+      };
+      const response = await request(TEST_URL)
+        .post("/v1/batch/scrape")
+        .set("Authorization", `Bearer ${process.env.TEST_API_KEY}`)
+        .set("Content-Type", "application/json")
+        .send(scrapeRequest);
+        
+      expect(response.statusCode).toBe(400);
+      expect(response.body.details[0].message).toContain("does not support social media scraping")
+    });
+
+    it.concurrent("should return a successful response with a valid API key", async () => {
+      const scrapeRequest: BatchScrapeRequest = {
+        urls: ["https://mendable.ai", "https://www.google.com", "https://example.com"],
+      };
+      const response = await request(TEST_URL)
+        .post("/v1/batch/scrape")
+        .set("Authorization", `Bearer ${process.env.TEST_API_KEY}`)
+        .set("Content-Type", "application/json")
+        .send(scrapeRequest);
+        
+      expect(response.statusCode).toBe(200);
+    });
+  });
+
+  describe("GET /v1/batch/scrape/:jobId:", () => {
+    it.concurrent("should require authorization", async () => {
+      const response = await request(TEST_URL).get("/v1/batch/scrape/123");
+      expect(response.statusCode).toBe(401);
+    });
+
+    it.concurrent("should return an error response with an invalid API key", async () => {
+      const response = await request(TEST_URL)
+        .get("/v1/batch/scrape/123")
+        .set("Authorization", `Bearer invalid-api-key`);
+      expect(response.statusCode).toBe(401);
+    });
+
+    it.concurrent(
+      "should return Job not found for invalid job ID",
+      async () => {
+        const response = await request(TEST_URL)
+          .get("/v1/batch/scrape/invalidJobId")
+          .set("Authorization", `Bearer ${process.env.TEST_API_KEY}`);
+        expect(response.statusCode).toBe(404);
+      }
+    );
+
+    it.concurrent(
+      "should return successful data for a valid batch scrape",
+      async () => {
+        const batchScrapeRequest: BatchScrapeRequest = {
+          urls: ["https://mendable.ai", "https://www.google.com", "https://example.com"],
```

Comment:
dont use mendable.ai, use firecrawl.dev instead. Also, use scrapethissite.com instead of google.